File: prime/src/integration-tests/java/org/ostelco/prime/TestPrimeConfig.java
package org.ostelco.prime;

import io.dropwizard.testing.DropwizardTestSupport;
import io.dropwizard.testing.ResourceHelpers;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.ostelco.prime.config.PrimeConfiguration;

import static org.junit.Assert.assertNotNull;

public class TestPrimeConfig {

    private static final DropwizardTestSupport<PrimeConfiguration> SUPPORT =
            new DropwizardTestSupport<>(PrimeApplication.class,
                    ResourceHelpers.resourceFilePath("config.yaml"));

    @BeforeClass
    public static void beforeClass() {
        SUPPORT.before();
    }

    @AfterClass
    public static void afterClass() {
        SUPPORT.after();
    }

    /**
     * Do nothing.
     * This test will just start and stop the server.
     * It will validate config file in 'src/test/resources/config.yaml'
     */
    @Test
    public void test() {
        assertNotNull(SUPPORT);
    }
}

File: prime/src/integration-tests/kotlin/org/ostelco/prime/TestPrimeConfig.kt
package org.ostelco.prime

import com.palantir.docker.compose.DockerComposeRule
import com.palantir.docker.compose.connection.waiting.HealthChecks
import io.dropwizard.testing.DropwizardTestSupport
import io.dropwizard.testing.ResourceHelpers
import org.joda.time.Duration
import org.junit.AfterClass
import org.junit.Assert.assertNotNull
import org.junit.BeforeClass
import org.junit.ClassRule
import org.junit.Test

class TestPrimeConfig {

    /**
     * Do nothing.
     * This test will just start and stop the server.
     * It will validate config file in 'src/integration-tests/resources/config.yaml'
     */
    @Test
    fun test() {
        assertNotNull(SUPPORT)
    }

    companion object {

        private val SUPPORT:DropwizardTestSupport<PrimeConfiguration> =
                DropwizardTestSupport(
                        PrimeApplication::class.java,
                        ResourceHelpers.resourceFilePath("config.yaml"))

        @ClassRule
        @JvmField
        var docker: DockerComposeRule = DockerComposeRule.builder()
                .file("src/integration-tests/resources/docker-compose.yaml")
                .waitingForService("neo4j", HealthChecks.toHaveAllPortsOpen())
                .waitingForService("neo4j",
                        HealthChecks.toRespond2xxOverHttp(7474) {
                            port -> port.inFormat("http://\$HOST:\$EXTERNAL_PORT/browser")
                        },
                        Duration.standardSeconds(20L))
                .build()

        @JvmStatic
        @BeforeClass
        fun beforeClass() {
            SUPPORT.before()
        }

        @JvmStatic
        @AfterClass
        fun afterClass() {
            SUPPORT.after()
        }
    }
}
